Framework/Algorithms/src/ChangePulsetime2.cpp

```diff
@@ -0,0 +1,63 @@
+#include "MantidAlgorithms/ChangePulsetime2.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidKernel/ArrayProperty.h"
+#include "MantidKernel/System.h"
+
+namespace Mantid {
+namespace Algorithms {
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(ChangePulsetime2)
+
+using namespace Mantid::Kernel;
+using namespace Mantid::API;
+using namespace Mantid::DataObjects;
+using std::size_t;
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void ChangePulsetime2::init() {
+  declareIndexProperty<EventWorkspace>("InputWorkspace");
```

Comment:
I think I agree here that it's nicer to only have to look in 1 place to see what you are declaring. I guess it also depends if there is an obvious choice for a default, I'm not sure that is the case here...

On the face of it here you have no idea that something else has been declared for you with some other type. As mentioned previously I think it would also be good to indicate that this declares more that 1 property.